spec/v2/payload_sanitizer_spec.rb

```diff
@@ -36,6 +36,10 @@ class BlankValuesPayloadSanitizer < InputSanitizer::V2::PayloadSanitizer
   url :non_blank_url, :allow_blank => false
 end
 
+class AllowProcPayloadSanitizer < InputSanitizer::V2::PayloadSanitizer
+  string :status, allow: -> { %w[current past] }
```

Comment:
The test case doesn't really show the dynamic nature of the change (by dynamic I mean the block is evaluated for each sanitization). The given test cases would still pass even if we'd memoize the return value of the Proc after first run.
My suggestion is to simulate the real scenario you're aiming at by manipulating the allowed set of values during the test case to prove that it works as expected.